web/src/base_page_params.ts

```diff
@@ -72,7 +72,18 @@ const stats_params_schema = default_params_schema.extend({
 // Sync this with corporate.views.portico.team_view.
 const team_params_schema = default_params_schema.extend({
     page_type: z.literal("team"),
-    contributors: z.unknown(),
+    contributors: z.optional(
+        z.array(
+            z
+                .object({
+                    avatar: z.string(),
+                    github_username: z.optional(z.string()),
+                    email: z.optional(z.string()),
+                    name: z.optional(z.string()),
+                })
+                .catchall(z.number()),
```

Comment:
Probably worth a comment explaining the need for `catchall`

Reply:
I was kinda confused about how to exactly type this. Since repo names will change or increase over time but the type will remain `string(repo_name) : number(contributions)` for repos. `catchall(z.number())`, is for parsing these keys and values.

Reply:
Yeah, the `catchall` makes sense but a comment near the `catchall` explaining the need would be nice IMO.

Reply:
Updated! Does this comment look okay?

Reply:
Now that I rethink about it, can't we use something like `z.record(z.string(), z.number())`  ??

Reply:
Does `and` work? Something like -
```diff
 web/src/base_page_params.ts | 7 +++----
 web/src/portico/team.ts     | 2 +-
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/web/src/base_page_params.ts b/web/src/base_page_params.ts
index bd163cd568..4aa792ff23 100644
--- a/web/src/base_page_params.ts
+++ b/web/src/base_page_params.ts
@@ -1,5 +1,6 @@
 import {z} from "zod";
 
+import {all_repository_names} from "./portico/team";
 import {narrow_term_schema, state_data_schema} from "./state_data";
 
 const t1 = performance.now();
@@ -70,6 +71,7 @@ const stats_params_schema = default_params_schema.extend({
 });
 
 // Sync this with corporate.views.portico.team_view.
+const repository = z.enum(all_repository_names);
 const team_params_schema = default_params_schema.extend({
     page_type: z.literal("team"),
     contributors: z.optional(
@@ -81,10 +83,7 @@ const team_params_schema = default_params_schema.extend({
                     email: z.optional(z.string()),
                     name: z.optional(z.string()),
                 })
-                // Repository names may change or increase over time,
-                // so we use this to parse the contributions of the user in
-                // the given repository instead of typing every name.
-                .catchall(z.number()),
+                .and(z.record(repository, z.number())),
         ),
     ),
 });
diff --git a/web/src/portico/team.ts b/web/src/portico/team.ts
index 12f10f982b..852dc9ad0b 100644
--- a/web/src/portico/team.ts
+++ b/web/src/portico/team.ts
@@ -8,7 +8,7 @@ import assert from "minimalistic-assert";
 // TODO: We can avoid this if we introduce a `contributions` object
 // referenced from Contributor, rather than having repository names be
 // direct keys in the namespace that also has `email`.
-const all_repository_names = [
+export const all_repository_names = [
     "docker-zulip",
     "errbot-backend-zulip",
     "github-actions-zulip",

```
This did not throw any `zod` error in local testing but I could not see any contributors in `http://localhost:9991/team/`. Does dev env has dummy contributors anyway??

Reply:
So first you have to pull the contributors list locally by doing `tools/fetch-contributor-data ` to test this. If you get rate limited use `github api_key`
You can find more about it here: https://zulip.readthedocs.io/en/latest/contributing/counting-contributions.html#relevant-source-code

Reply:
Also, the suggested approach doesn't work :(

Reply:
Alas, then we should go forward with `catchall` I guess.